src/platform_impl/linux/wayland/window/state.rs

```diff
@@ -901,6 +928,10 @@ impl Drop for WindowState {
         }
 
         surface.destroy();
+
+        if (self.blur.is_some()) {
+            self.blur.take().unwrap().release();
+        }
```

Comment:
And do that before the `surface.destroy`.

```suggestion
        if let Some(blur) = self.blur.take() {
            blur.release();
        }
```